reyna/src/com/b2msolutions/reyna/services/BlackoutTime.java

```diff
@@ -0,0 +1,60 @@
+package com.b2msolutions.reyna.services;
+
+import android.content.Context;
+import com.b2msolutions.reyna.Preferences;
+import com.b2msolutions.reyna.Time;
+import com.b2msolutions.reyna.TimeRange;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+public class BlackoutTime {
+
+    private final Context context;
+
+    public BlackoutTime(Context context) {
+        this.context = context;
+    }
+
+    public boolean canSendOnWlan(Calendar now) throws ParseException {
+        Preferences preferences = new Preferences(context);
+        String range = preferences.getWlanRange();
+        return canSendAtTime(now, range);
+    }
+
+    public boolean canSendOnWwan(Calendar now) throws ParseException {
+        Preferences preferences = new Preferences(context);
+        String range = preferences.getWwanRange();
+        return canSendAtTime(now, range);
+    }
+
+    private boolean canSendAtTime(Calendar now, String range) throws ParseException {
+        if (range.length() == 0) return true;
+
+        String[] rangesSplit = range.split(",");
+        for (String rangeSplit : rangesSplit) {
+            List<Time> times = parseTime(rangeSplit);
```

Comment:
in case parseTime throws, this should return true.

or can be handled in the higher canSend caller.
we need to handle invalid configuration values
@Youhana-Hana